Refactor knative related helm charts (#3187)
kyma-project/kyma: resources/knative-eventing/README.md

# Knative Eventing

## Overview

> This chart packs the [knative eventing](https://github.com/knative/docs/tree/master/docs/eventing) release files.

Reviewer: ```suggestion
This chart includes [knative eventing](https://github.com/knative/docs/tree/master/docs/eventing) release files.
```

# Knative Eventing

## Overview

This chart includes [knative eventing](https://github.com/knative/docs/tree/master/docs/eventing) release files.

Included releases:
 * https://github.com/knative/eventing/releases/download/v0.4.1/release.yaml

Kyma-specific changes:
 * Every CRD has the `helm.sh/hook: crd-install` annotation set. This forces Helm to install the CRDs before other resources.

>  * The `default-channel-config` is set to empty since when this chart is installed, we do not have a provisioner ready.

Reviewer: ```suggestion
 * The `default-channel-config` is left empty since when this chart is installed, there is no provisioner ready.
```

>  * The creation of `knative-eventing` Namespace has been removed as this will taken care by installation process.

Reviewer: ```suggestion
 * The `knative-eventing` Namespace is no longer created. This happens during the installation process.
```

>  * The `in-memory-channel` has been removed as Kyma use NATS Streaming based provisioner OOTB.

Reviewer: ```suggestion
 * The `in-memory-channel` no longer exists, as Kyma uses NATS Streaming-based provisioner out of the box.
```

# Knative Eventing

## Overview

This chart includes [knative eventing](https://github.com/knative/docs/tree/master/docs/eventing) release files.

Included releases:
 * https://github.com/knative/eventing/releases/download/v0.4.1/release.yaml

Kyma-specific changes:
 * Every CRD has the `helm.sh/hook: crd-install` annotation set. This forces Helm to install the CRDs before other resources.

>  * The `default-channel-config` is left empty since when this chart is installed, there is no provisioner ready.

Reviewer: ```suggestion
 * The `default-channel-config` is left empty since there is no provisioner ready during the chart installation.
```
